src/main/java/io/cryostat/net/web/http/api/beta/RecordingMetadataPatchBodyHandler.java

```diff
@@ -0,0 +1,91 @@
+/*
+ * Copyright The Cryostat Authors
+ *
+ * The Universal Permissive License (UPL), Version 1.0
+ *
+ * Subject to the condition set forth below, permission is hereby granted to any
+ * person obtaining a copy of this software, associated documentation and/or data
+ * (collectively the "Software"), free of charge and under any and all copyright
+ * rights in the Software, and any and all patent rights owned or freely
+ * licensable by each licensor hereunder covering either (i) the unmodified
+ * Software as contributed to or provided by such licensor, or (ii) the Larger
+ * Works (as defined below), to deal in both
+ *
+ * (a) the Software, and
+ * (b) any piece of software and/or hardware listed in the lrgrwrks.txt file if
+ * one is included with the Software (each a "Larger Work" to which the Software
+ * is contributed by such licensors),
+ *
+ * without restriction, including without limitation the rights to copy, create
+ * derivative works of, display, perform, and distribute the Software and make,
+ * use, sell, offer for sale, import, export, have made, and have sold the
+ * Software and the Larger Work(s), and to sublicense the foregoing rights on
+ * either these or other terms.
+ *
+ * This license is subject to the following condition:
+ * The above copyright notice and either this complete permission notice or at
+ * a minimum a reference to the UPL must be included in all copies or
+ * substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+package io.cryostat.net.web.http.api.beta;
+
+import java.util.Set;
+
+import javax.inject.Inject;
+
+import io.cryostat.net.AuthManager;
+import io.cryostat.net.security.ResourceAction;
+import io.cryostat.net.web.http.AbstractAuthenticatedRequestHandler;
+import io.cryostat.net.web.http.api.ApiVersion;
+
+import io.vertx.core.http.HttpMethod;
+import io.vertx.ext.web.RoutingContext;
+import io.vertx.ext.web.handler.BodyHandler;
+
+class RecordingMetadataPatchBodyHandler extends AbstractAuthenticatedRequestHandler {
+
+    static final BodyHandler BODY_HANDLER = BodyHandler.create(true).setHandleFileUploads(false);
+
+    @Inject
+    RecordingMetadataPatchBodyHandler(AuthManager auth) {
+        super(auth);
+    }
+
+    @Override
+    public ApiVersion apiVersion() {
+        return ApiVersion.BETA;
+    }
+
+    @Override
+    public int getPriority() {
+        return DEFAULT_PRIORITY - 1;
+    }
+
+    @Override
+    public HttpMethod httpMethod() {
+        return HttpMethod.PATCH;
```

Comment:
Along with this I would also propose that the `HyperlinkedSerializableRecordingDescriptor` and any other response objects that contain these new labels should contain a more generalized "metadata" field, which currently contains just the one nested "labels" field. This should be more extensible for the future.

Reply:
Yeah, `POST` makes sense. I was originally considering the labels as part of a recording state, but since the existing `PATCH` handlers can create new labels, it makes sense to rename them to `POST`.

For your `GET` handler examples, did you mean to add handlers specifically for metadata? As of now, the labels are returned with the recording `GET` handlers.

Reply:
I'm still tossing around the idea of `GET`s for `/metadata` and `/metadata/labels`. If we don't have these then the only way to get metadata for a recording is to `GET /recordings`/`GET /targets/:targetId/recordings` and then filter that result by the desired name on the client side. On the other hand, I don't like that this creates an implication or expectation that the API URLs are hierarchical in a way that totally mirrors the JSON response structures.

Really, the root problem here is that I initially designed the `GET /api/v1//targets/:targetId/recordings/:recordingName` to respond with the JFR binary file directly, and not the recording descriptor object. The binary should have been sent either by making a request on a different endpoint, or by the client setting the request header `Content-Type`, or something along those lines. This fix is way out of scope of this current PR though.

I suppose since we have no direct use case of our own for the `GET /metadata` endpoints, let's skip implementing them for now. Maybe they make sense to do later, or maybe we never do them and instead make API v2.2+ endpoints that re-implement the v1 APIs in a smarter way, which solves the recording metadata problem at the same time.

Reply:
To clarify, even without a `GET /metadata` being implemented, I would still like the serialized recording descriptor to nest the `labels` field within a `metadata` field.

```json
{
  "name": "myRecording",
  "state": "RUNNING",
  "metadata": {
    "labels": {
      "myApp": "cryostat"
    }
  }
}
```

On the Java side this probably just means add a simple inner static class within the descriptor class that just holds a reference to the labels map, creating this nested structure.